### DISCUSSION TITLE
Add a feature to filter accidental key presses in mechanical keyboard, also known as chattering

### DISCUSSION THREAD

**Discussion by gustarmartins** [ppy/osu#16920] [Ideas]
I think the title is self-explanatory. Nowadays most people will go after third-party softwares to block the chattering caused by worn-out mechanical switches (which is not very rare to happen, especially for games like osu!), but it would be really good if you could add this feature natively on the game. 

The reason for that is to avoid people from having to install third-party software that can harm their PC or cause incompatibilities with game's anti-cheats for the behaviour of modifying the input. 

Chattering is really bad for osu! because it will definitely make you miss on zig-zag patterns, stacks, and such.

**Comment by bdach** [comment]
Not sure if this will happen but just going to note that if done in software, then this feature would cause increased input lag, because there is no other way to watch out for false presses than to monitor them for a few input frames and then discard anything that is too short to be a press.

**Reply by gustarmartins** [reply]
I wonder how big is the input lag and why not add it anyway? I'd rather play with ''input lag'' than with my keyboard chattering.

**Reply by gustarmartins** [reply]
The input lag would affect only users who chooses to enable the feature anyways.

**Comment by Gamingzone123** [comment]
kind of a hardware issue, get new switches or a new keyboard (mechanical switches are typically hard-wearing anyway if this is happening your keyboard is probably starting to wear out on the whole), and as bdach said, input latency if done in osu
